fdbkubernetesmonitor/kubernetes.go

```diff
@@ -204,78 +225,79 @@ func (podClient *PodClient) UpdateAnnotations(monitor *Monitor) error {
 			APIVersion: "v1",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace:   podClient.metadata.Namespace,
-			Name:        podClient.metadata.Name,
+			Namespace:   podClient.podMetadata.Namespace,
+			Name:        podClient.podMetadata.Name,
 			Annotations: annotations,
 		},
 	}, client.Apply, client.FieldOwner("fdb-kubernetes-monitor"), client.ForceOwnership)
 }
 
 // OnAdd is called when an object is added.
 func (podClient *PodClient) OnAdd(obj interface{}) {
-	pod, ok := obj.(*corev1.Pod)
-	if !ok {
-		return
-	}
-
-	podClient.Logger.Info("Got event for OnAdd", "name", pod.Name, "namespace", pod.Namespace)
-	podClient.metadata = &metav1.PartialObjectMetadata{
-		TypeMeta:   pod.TypeMeta,
-		ObjectMeta: pod.ObjectMeta,
+	switch castedObj := obj.(type) {
+	case *corev1.Pod:
+		podClient.Logger.Info("Got event for OnAdd for Pod resource", "name", castedObj.Name, "namespace", castedObj.Namespace)
+		podClient.podMetadata = &metav1.PartialObjectMetadata{
+			TypeMeta:   castedObj.TypeMeta,
+			ObjectMeta: castedObj.ObjectMeta,
+		}
+	case *corev1.Node:
+		podClient.Logger.Info("Got event for OnAdd Node resource", "name", castedObj.Name)
+		podClient.podMetadata = &metav1.PartialObjectMetadata{
```

Comment:
In the `CreatePodClient` function we set the node metadata on podClient.nodeMetadata. Should we be doing that here as well?

Reply:
I added unit tests for this functionality.